express-server/src/types/context.ts

```diff
@@ -50,24 +67,13 @@ export const env = z.object({
   FIREBASE_DATABASE_URL: z
     .string({ required_error: "Missing FIREBASE_DATABASE_URL" })
     .url({ message: "FIREBASE_DATABASE_URL in env should be a valid url" }),
-  // REDIS_HOST: z.string({ required_error: "Missing REDIS_HOST" }),
```

Comment:
there's a subtle dev/prod difference hiding here:
- in dev, we use REDIS_HOST + REDIS_PORT (pointing to a local process)
- in prod, we use REDIS_URL (pointing to a remote instance)
we can't get rid of this without accounting for both / merging these different config approaches